cfme/cloud/instance/__init__.py

```diff
@@ -306,54 +303,133 @@ def power_control_from_cfme(self, *args, **kwargs):
+        image = self.appliance.rest_api.collections.templates.filter(
+            Q('name', '=', image_name) & Q('ems_id', '=', provider_rest.id))[0]
+        # find out flavor
+        if ':' in provisioning['instance_type'] and self.provider.one_of(EC2Provider):
+            instance_type = provisioning['instance_type'].split(':')[0].strip()
         else:
-            view.form.submit_button.click()
+            instance_type = provisioning['instance_type']
+        flavor = self.appliance.rest_api.collections.flavors.filter(
+            Q('name', '=', instance_type) & Q('ems_id', '=', provider_rest.id))[0]
+        # find out cloud network
+        cloud_network_name = provisioning.get('cloud_network').strip()
+        if self.provider.one_of(EC2Provider, AzureProvider):
+            cloud_network_name = cloud_network_name.split()[0]
+        cloud_network = self.appliance.rest_api.collections.cloud_networks.filter(
+            Q('name', '=', cloud_network_name) & Q('enabled', '=', 'true'))[0]
+        # find out cloud subnet
+        cloud_subnet = self.appliance.rest_api.collections.cloud_subnets.filter(
+            Q('cloud_network_id', '=', cloud_network['id']))[0]
+        # find out availability zone
+        availability_zone_id = None
+        av_zone_name = provisioning.get('availability_zone')
+        if av_zone_name:
+            availability_zone_id = self.appliance.rest_api.collections.availability_zones.filter(
+                Q('name', '=', av_zone_name) & Q('ems_id', '=', flavor.ems_id))[0].id
+        # find out cloud tenant
+        tenant_name = provisioning.get('cloud_tenant')
+        if tenant_name:
+            tenant = self.appliance.rest_api.collections.cloud_tenants.filter(
+                Q('name', '=', tenant_name) &
+                Q('ems_id', '=', provider_rest.id) &
+                Q('enabled', '=', 'true'))[0]
+
+        resource_group_id = None
+        if self.provider.one_of(AzureProvider):
+            resource_groups = self.appliance.rest_api.get(
+                '{}?attributes=resource_groups'.format(provider_rest._href))['resource_groups']
+            resource_group_id = None
+            resource_group_name = provisioning.get('resource_group')
+            for res_group in resource_groups:
+                if (res_group['name'] == resource_group_name and
+                        res_group['ems_id'] == provider_rest.id):
+                    resource_group_id = res_group['id']
+                    break
+
+        inst_args = {
+            "version": "1.1",
+            "template_fields": {
+                "guid": image.guid
+            },
+            "vm_fields": {
+                "placement_auto": False,
+                "vm_name": self.name,
+                "instance_type": flavor['id'],
+                "request_type": "template",
+                "cloud_network": cloud_network['id'],
+                "cloud_subnet": cloud_subnet['id'],
```

Comment:
What about "placement_availability_zone" and "security_groups" that were present in the original code?

Reply:
they are not mandatory. we can provision w/o those. availability_zone is added on providers that have it in yamls